lambdas/retrieve-cri-credential/src/main/java/uk/gov/di/ipv/core/retrievecricredential/RetrieveCriCredentialHandler.java

```diff
@@ -4,12 +4,234 @@
+        this.ciStorageService = new CiStorageService(configurationService);
+    }
+
     @Override
     public APIGatewayProxyResponseEvent handleRequest(
             APIGatewayProxyRequestEvent input, Context context) {
-        return null;
+        LogHelper.attachComponentIdToLogs();
+        IpvSessionItem ipvSessionItem = null;
+
+        RetrieveCriRequestDto request =
+                RequestHelper.convertRequest(input, RetrieveCriRequestDto.class);
+
+        try {
+            String ipvSessionId = RequestHelper.getIpvSessionId(input);
+            ipvSessionItem = ipvSessionService.getIpvSession(ipvSessionId);
+            ClientSessionDetailsDto clientSessionDetailsDto =
+                    ipvSessionItem.getClientSessionDetails();
+            String userId = clientSessionDetailsDto.getUserId();
+
+            LogHelper.attachGovukSigninJourneyIdToLogs(
+                    clientSessionDetailsDto.getGovukSigninJourneyId());
+
+            AuditEventUser auditEventUser =
+                    new AuditEventUser(
+                            userId,
+                            ipvSessionId,
+                            clientSessionDetailsDto.getGovukSigninJourneyId());
+            this.componentId =
+                    configurationService.getSsmParameter(
+                            ConfigurationVariable.AUDIENCE_FOR_CLIENTS);
+
+            CredentialIssuerConfig credentialIssuerConfig =
+                    configurationService.getCredentialIssuer(request.getCredentialIssuerId());
+
+            String apiKey =
+                    configurationService.getCriPrivateApiKey(credentialIssuerConfig.getId());
+
+            List<SignedJWT> verifiableCredentials =
+                    credentialIssuerService.getVerifiableCredential(
+                            BearerAccessToken.parse(request.getAccessToken()),
+                            credentialIssuerConfig,
+                            apiKey);
+
+            for (SignedJWT vc : verifiableCredentials) {
+                verifiableCredentialJwtValidator.validate(vc, credentialIssuerConfig, userId);
+
+                sendIpvVcReceivedAuditEvent(auditEventUser, vc);
+
+                if (configurationService.isNotRunningInProd()) {
+                    LOGGER.info("Submitting VC to CI storage system");
+                    submitVCAndSwallowErrors(vc, clientSessionDetailsDto.getGovukSigninJourneyId());
+                }
+
+                credentialIssuerService.persistUserCredentials(
+                        vc.serialize(), request.getCredentialIssuerId(), userId);
+            }
+
+            updateVisitedCredentials(ipvSessionItem, request.getCredentialIssuerId(), true, null);
+
+            return ApiGatewayResponseGenerator.proxyJsonResponse(
+                    HttpStatus.SC_OK,
+                    new JourneyResponse("/journey/cri/validate/" + credentialIssuerConfig.getId()));
```

Comment:
I think this might need to be /journey/next.

That's what we currently have in the other lambda I think. I can't quite remember, but I have a feeling we got rid of the validate lambda